mlir/lib/Transforms/Utils/GreedyPatternRewriteDriver.cpp

```diff
@@ -462,18 +463,38 @@ bool GreedyPatternRewriteDriver::processWorklist() {
           // Materialize Attributes as SSA values.
           Operation *constOp = op->getDialect()->materializeConstant(
               *this, ofr.get<Attribute>(), resultType, op->getLoc());
```

Comment:
`materializeConstant` will send a `notifyOperationInserted` which is not correct if we erase the ops again. We need a new temporary `OpBuilder` instead of `*this` and if all constants materialized successfully, trigger a manual `this->notifyOperationInserted` for each materialized constant.

Reply:
Actually, you are erasing the ops with `eraseOp(op);`, which will send a `notifyOperationRemoved`. So it should be fine.